resources/qml/Settings/SettingView.qml

```diff
@@ -9,229 +9,406 @@ import QtQuick.Layouts 1.1
+            property bool lastFindingSettings: false
+
+            onTextChanged:
             {
-                target: provider
-                property: "containerStackId"
-                when: model.settable_per_extruder || (inheritStackProvider.properties.limit_to_extruder != null && inheritStackProvider.properties.limit_to_extruder >= 0);
-                value:
+                definitionsModel.filter = {"label": "*" + text};
+                findingSettings = (text.length > 0);
+                if(findingSettings != lastFindingSettings)
                 {
-                    if(!model.settable_per_extruder || machineExtruderCount.properties.value == 1)
-                    {
-                        //Not settable per extruder or there only is global, so we must pick global.
-                        return Cura.MachineManager.activeMachineId;
-                    }
-                    if(inheritStackProvider.properties.limit_to_extruder != null && inheritStackProvider.properties.limit_to_extruder >= 0)
+                    if(findingSettings)
                     {
-                        //We have limit_to_extruder, so pick that stack.
-                        return ExtruderManager.extruderIds[String(inheritStackProvider.properties.limit_to_extruder)];
+                        expandedCategories = definitionsModel.expanded.slice();
+                        definitionsModel.expanded = ["*"];
+                        definitionsModel.showAncestors = true;
+                        definitionsModel.showAll = true;
                     }
-                    if(ExtruderManager.activeExtruderStackId)
+                    else
                     {
-                        //We're on an extruder tab. Pick the current extruder.
-                        return ExtruderManager.activeExtruderStackId;
+                        definitionsModel.expanded = expandedCategories;
+                        definitionsModel.showAncestors = false;
+                        definitionsModel.showAll = false;
                     }
-                    //No extruder tab is selected. Pick the global stack. Shouldn't happen any more since we removed the global tab.
-                    return Cura.MachineManager.activeMachineId;
+                    lastFindingSettings = findingSettings;
                 }
             }
 
-            // Specialty provider that only watches global_inherits (we cant filter on what property changed we get events
-            // so we bypass that to make a dedicated provider).
-            UM.SettingPropertyProvider
+            Keys.onEscapePressed:
             {
-                id: inheritStackProvider
-                containerStackId: Cura.MachineManager.activeMachineId
-                key: model.key
-                watchedProperties: [ "limit_to_extruder" ]
+                filter.text = "";
             }
+        }
 
-            UM.SettingPropertyProvider
+        MouseArea
+        {
+            id: mouseArea
```

Comment:
It would do us well to give this a better name. I have no idea what this mouse area is used for.